src/libImaging/BcnDecode.c

```diff
@@ -750,8 +750,10 @@ decode_bc6_block(rgb32f *col, const UINT8 *src, int sign) {
         }
     }
     if (info->tr) { /* apply deltas */
-        for (i = 3; i < numep; i++) {
-            endpoints[i] = (endpoints[i] + endpoints[0]) & mask;
+        for (i = 3; i < numep; i += 3) {
+            endpoints[i + 0] = (endpoints[i + 0] + endpoints[0]) & mask;
```

Comment:
```suggestion
            endpoints[i] = (endpoints[i] + endpoints[0]) & mask;
```

Reply:
I wanted to keep the code style the same, but okay.